src/diem/testing/miniwallet/app/app.py

```diff
@@ -172,6 +185,36 @@ def _send_external_payment_txn(self, txn: Transaction) -> None:
             self._start_external_payment_txn(txn)
 
     def _start_external_payment_txn(self, txn: Transaction) -> None:
+        if identifier.diem_id.is_diem_id(str(txn.payee)):
+            reference_id = str(uuid.uuid4())
+            self.store.create(ReferenceID, account_id=txn.account_id, reference_id=reference_id)
```

Comment:
add before create validation to ensure uniq:
```
before: lambda _: self.validate_uniq_reference_id(reference_id)
```
`validate_uniq_reference_id` should raise error if not.
put the self.store.create(ReferenceID... in a loop to create until success.